acvm-repo/acvm/tests/solver.rs

```diff
@@ -722,3 +724,531 @@ fn memory_operations() {
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_mul_one_r((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_mul_op(), modulus, xs, one);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_sub_self((xs, modulus) in bigint_with_modulus()) {
+        let expected_results = drop_use_constant(&bigint_zeroed(&xs));
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs.clone(), xs);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_sub_zero((xs, modulus) in bigint_with_modulus()) {
+        let zero = bigint_zeroed(&xs);
+        let expected_results: Vec<_> = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs, zero);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_sub_one((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results: Vec<_> = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_sub_op(), modulus, xs, one);
+        prop_assert!(results != expected_results, "{:?} == {:?}", results, expected_results)
+    }
+
+    #[test]
+    fn bigint_div_self((xs, modulus) in bigint_with_modulus()) {
+        let one = drop_use_constant(&bigint_to_one(&xs));
+        let results = bigint_solve_binary_op(bigint_div_op(), modulus, xs.clone(), xs);
+        prop_assert_eq!(results, one)
+    }
+
+    #[test]
+    fn bigint_div_by_zero((xs, modulus) in bigint_with_modulus()) {
+        let zero = bigint_zeroed(&xs);
+        let expected_results = drop_use_constant(&zero);
+        let results = bigint_solve_binary_op(bigint_div_op(), modulus, xs, zero);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_div_one((xs, modulus) in bigint_with_modulus()) {
+        let one = bigint_to_one(&xs);
+        let expected_results = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_div_op(), modulus, xs, one);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_div_zero((xs, modulus) in bigint_with_modulus()) {
```

Comment:
what is the difference with bigint_div_by_zero ?

Reply:
This is testing that `0 / x == 0`